In [1]:
import pandas as pd
import matplotlib.pyplot as plt

# Problem 2: Leukemia Diagnosis

In [2]:
# load the data
url = 'https://raw.githubusercontent.com/um-perez-alvaro/Data-Science-Practice/master/Data/leukemia.csv'
data = pd.read_csv(url)
data.head()

HTTPError: HTTP Error 404: Not Found

In [16]:
data.shape

(198, 16064)

This dataset contains cancer gene-expression levels (16,063 genes, 198 samples) from the paper *Multiclass cancer diagnosis using tumor gene expression signatures (Golub et all, 2001)*

Cancer classes are labelled as follows:


- 1: breast
- 2:  prostate
- 3:  lung
- 4:  collerectal
- 5:  lymphoma
- 6:  bladder
- 7:  melanoma
- 8:  uterus
- 9:  **leukemia**
- 10: renal
- 11: pancreas
- 12: ovary
- 13: meso
- 14: cns


Your goal is to train a regression model to classify cancers as either leukemia or not-leukemia

**Part 1:** Replace the `label` values with:

$$
\left\{ \begin{array}{ll} 1 & \mbox{ if cancer is leukemia}\\
0 & \mbox{ if cancer is not leukemia}. \end{array}\right.
$$

**Part 2:** Define X and y from the DataFrame, and then split X and y into training and testing sets

**Part 3:** Use the training set to train a **Lasso regression model**. 
Plot the model's coefficients. How many coefficients are equal to 0?

**Part 4:**  Use your regression model to classify all the cancers in the test set.  
Use the rule:

$$
\mbox{classify cancer as a }\left\{ \begin{array}{ll} 
\mbox{leukemia} & \mbox{ if y_test_pred}>0.5 \\
\mbox{not leukemia }5 & \mbox{ if predicted y_test_pred}\leq 0.5
\end{array}\right.
$$



How many cancers are misclassified? 